engine-tests/src/tests/one_inch.rs

```diff
@@ -104,7 +104,7 @@ fn test_1_inch_limit_order_deploy() {
     // at least 45% of which is from wasm execution
     let wasm_fraction = 100 * profile.wasm_gas() / profile.all_gas();
     assert!(
-        (45..=55).contains(&wasm_fraction),
+        (55..=60).contains(&wasm_fraction),
```

Comment:
This increase in the wasm fraction was a red-flag to me. If the new version of the rust compiler was making the Wasm code more efficient then the fraction of gas spent on Wasm computation should go down, not up. The reason it goes up is because of now missing the touching trie node costs (see comment on `MockedExternalWithTrie::get_trie_nodes_count`). Please change this back to `45..=55` and revisit after resolving that comment.